src/plugins/intel_cpu/src/mkldnn_plugin.cpp

```diff
@@ -753,9 +883,72 @@ Parameter Engine::GetMetric(const std::string& name, const std::map<std::string,
         IE_SET_METRIC_RETURN(RANGE_FOR_STREAMS, range);
     } else if (name == METRIC_KEY(IMPORT_EXPORT_SUPPORT)) {
         IE_SET_METRIC_RETURN(IMPORT_EXPORT_SUPPORT, true);
-    } else {
-        IE_THROW() << "Unsupported metric key " << name;
     }
+
+    IE_THROW() << "Unsupported metric key " << name;
+}
+
+Parameter Engine::GetMetric(const std::string& name, const std::map<std::string, Parameter>& options) const {
+    if (isLegacyAPI())
+        return GetMetricLegacy(name, options);
+
+    auto RO_property = [](const std::string& propertyName) {
+        return ov::PropertyName(propertyName, ov::PropertyMutability::RO);
+    };
+    auto RW_property = [](const std::string& propertyName) {
+        return ov::PropertyName(propertyName, ov::PropertyMutability::RW);
+    };
+
+    if (name == ov::supported_properties) {
+        std::vector<ov::PropertyName> roProperties {RO_property(ov::available_devices.name()),
+                                                    RO_property(ov::supported_properties.name()),
+                                                    RO_property(ov::range_for_async_infer_requests.name()),
+                                                    RO_property(ov::range_for_streams.name()),
+                                                    RO_property(ov::device::full_name.name()),
+                                                    RO_property(ov::device::capabilities.name())
+        };
+        // the whole config is RW before network is loaded.
+        std::vector<ov::PropertyName> rwProperties {RW_property(ov::streams::num.name()),
+                                                    RW_property(ov::affinity.name()),
+                                                    RW_property(ov::inference_num_threads.name()),
+                                                    RW_property(ov::enable_profiling.name()),
+                                                    RW_property(ov::hint::inference_precision.name()),
+                                                    RW_property(ov::hint::performance_mode.name()),
+                                                    RW_property(ov::hint::num_requests.name()),
+                                                    RW_property(ov::cache_dir.name()),
```

Comment:
I guess we discussed that cache_dir shouldn't be explicitly returned by the plugin.

Reply:
Done